src/scitokens_internal.cpp

```diff
@@ -39,6 +39,94 @@ namespace scitokens {
 
 namespace internal {
 
+// BackgroundRefreshManager implementation
+void BackgroundRefreshManager::start() {
+    if (m_running) {
+        return;  // Already running
+    }
+
+    std::call_once(m_start_once, [this]() {
```

Comment:
This means the refresh can only be start/stopped once.  Make it so start/stop can be called multiple times (will use this in unit tests, for example) but keep it thread-safe, serializing the start/stop.